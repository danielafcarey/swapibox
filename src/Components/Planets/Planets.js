import React from 'react';
import PropTypes from 'prop-types';

import Card from '../Card/Card.js';

const Planets = ({ cardData, toggleFavorite }) => {
  const planetsCards = cardData.map(card => {
    const { id, favorite, name, population, climate, terrain, residents } = card;

    return (
      <Card
        id={ id }
        title={ name }
        item1={ `Population: ${population}` }
        item2={ `Climate: ${climate}` }
        item3={ `Terrain: ${terrain}` }
        item4={ `Residents: ${residents.join(', ')}` }
        favorite={ favorite }
        toggleFavorite={ toggleFavorite }
        key={ id }
      />
    )
  })

  return (
    <div className="card-container Planets">
      { planetsCards } 
    </div>
  );
}

Planets.propTypes = {
  cardData: PropTypes.arrayOf(PropTypes.object).isRequired
}


export default Planets;
