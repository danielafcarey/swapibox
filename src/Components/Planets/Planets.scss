.Planets{
  .Card {
    background: rgba(43, 43, 43, 0.65) url('../../images/planets.png') no-repeat center;
  }
}
