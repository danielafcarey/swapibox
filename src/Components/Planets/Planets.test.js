import React from 'react';
import { shallow } from 'enzyme';
import Planets from './Planets';

describe('Planets', () => {

  it('matches the snapshot with one card', () => {
    const mockCardData = [{
      id: 1,
      favorite: false,
      name: 'Daniela',
      climate: 'hot',
      population: 7000000000,
      residents: ['pupper'],
      terrain: 'rocks'
    }];
    const wrapper = shallow(
      <Planets
        cardData={ mockCardData }
        toggleFavorite={ jest.fn() }
      /> 
    );

    expect(wrapper).toMatchSnapshot();
  })

  it('matches the snapshot with cards', () => {
    const mockCardData = [{
      id: 1,
      favorite: false,
      name: 'Daniela',
      climate: 'hot',
      population: 7000000000,
      residents: ['pupper'],
      terrain: 'rocks'
    },
      {
        id: 1,
        favorite: false,
        name: 'Daniela',
        climate: 'hot',
        population: 7000000000,
        residents: ['pupper'],
        terrain: 'rocks'
      }
    ];
    const wrapper = shallow(
      <Planets
        cardData={ mockCardData }
        toggleFavorite={ jest.fn() }
      /> 
    );

    expect(wrapper).toMatchSnapshot();
  })

});

