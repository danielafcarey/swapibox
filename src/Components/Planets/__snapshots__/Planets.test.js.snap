// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planets matches the snapshot with cards 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Planets
    cardData={
        Array [
            Object {
              "climate": "hot",
              "favorite": false,
              "id": 1,
              "name": "Daniela",
              "population": 7000000000,
              "residents": Array [
                "pupper",
              ],
              "terrain": "rocks",
            },
            Object {
              "climate": "hot",
              "favorite": false,
              "id": 1,
              "name": "Daniela",
              "population": 7000000000,
              "residents": Array [
                "pupper",
              ],
              "terrain": "rocks",
            },
          ]
    }
    toggleFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          favorite={false}
          id={1}
          item1="Population: 7000000000"
          item2="Climate: hot"
          item3="Terrain: rocks"
          item4="Residents: pupper"
          title="Daniela"
          toggleFavorite={[Function]}
/>,
        <Card
          favorite={false}
          id={1}
          item1="Population: 7000000000"
          item2="Climate: hot"
          item3="Terrain: rocks"
          item4="Residents: pupper"
          title="Daniela"
          toggleFavorite={[Function]}
/>,
      ],
      "className": "card-container Planets",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "favorite": false,
          "id": 1,
          "item1": "Population: 7000000000",
          "item2": "Climate: hot",
          "item3": "Terrain: rocks",
          "item4": "Residents: pupper",
          "title": "Daniela",
          "toggleFavorite": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "favorite": false,
          "id": 1,
          "item1": "Population: 7000000000",
          "item2": "Climate: hot",
          "item3": "Terrain: rocks",
          "item4": "Residents: pupper",
          "title": "Daniela",
          "toggleFavorite": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            favorite={false}
            id={1}
            item1="Population: 7000000000"
            item2="Climate: hot"
            item3="Terrain: rocks"
            item4="Residents: pupper"
            title="Daniela"
            toggleFavorite={[Function]}
/>,
          <Card
            favorite={false}
            id={1}
            item1="Population: 7000000000"
            item2="Climate: hot"
            item3="Terrain: rocks"
            item4="Residents: pupper"
            title="Daniela"
            toggleFavorite={[Function]}
/>,
        ],
        "className": "card-container Planets",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "favorite": false,
            "id": 1,
            "item1": "Population: 7000000000",
            "item2": "Climate: hot",
            "item3": "Terrain: rocks",
            "item4": "Residents: pupper",
            "title": "Daniela",
            "toggleFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "favorite": false,
            "id": 1,
            "item1": "Population: 7000000000",
            "item2": "Climate: hot",
            "item3": "Terrain: rocks",
            "item4": "Residents: pupper",
            "title": "Daniela",
            "toggleFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Planets matches the snapshot with one card 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Planets
    cardData={
        Array [
            Object {
              "climate": "hot",
              "favorite": false,
              "id": 1,
              "name": "Daniela",
              "population": 7000000000,
              "residents": Array [
                "pupper",
              ],
              "terrain": "rocks",
            },
          ]
    }
    toggleFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          favorite={false}
          id={1}
          item1="Population: 7000000000"
          item2="Climate: hot"
          item3="Terrain: rocks"
          item4="Residents: pupper"
          title="Daniela"
          toggleFavorite={[Function]}
/>,
      ],
      "className": "card-container Planets",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "favorite": false,
          "id": 1,
          "item1": "Population: 7000000000",
          "item2": "Climate: hot",
          "item3": "Terrain: rocks",
          "item4": "Residents: pupper",
          "title": "Daniela",
          "toggleFavorite": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            favorite={false}
            id={1}
            item1="Population: 7000000000"
            item2="Climate: hot"
            item3="Terrain: rocks"
            item4="Residents: pupper"
            title="Daniela"
            toggleFavorite={[Function]}
/>,
        ],
        "className": "card-container Planets",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "favorite": false,
            "id": 1,
            "item1": "Population: 7000000000",
            "item2": "Climate: hot",
            "item3": "Terrain: rocks",
            "item4": "Residents: pupper",
            "title": "Daniela",
            "toggleFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
